src/vs/workbench/api/common/menusExtensionPoint.ts

```diff
@@ -347,10 +347,16 @@ commandsExtensionPoint.setHandler(extensions => {
 
 		const { icon, enablement, category, title, command } = userFriendlyCommand;
 
+		let iconClassName: string | undefined;
 		let absoluteIcon: { dark: URI; light?: URI; } | undefined;
 		if (icon) {
 			if (typeof icon === 'string') {
-				absoluteIcon = { dark: resources.joinPath(extension.description.extensionLocation, icon) };
+				if (icon.indexOf('icon://vscode.codicons/') === 0) {
+					iconClassName = escape(`codicon-${URI.parse(icon).path.substr(1)}`);
```

Comment:
@jrieken Not sure if I should have bothered with the `URI.parse` here, but wanted to keep it consistent with the other usages